### PR TITLE
[HttpKernel] recover from failed deserializations

### LINKED ISSUE #42031: [HttpCache] Invalid Store::getMetadata() return type
**Symfony version(s) affected**: 4.4.24

**Description**
I frequently get the following error in my prod server logs : `Return value of Symfony\\Component\\HttpKernel\\HttpCache\\Store::getMetadata() must be of the type array, bool returned`

**How to reproduce**
I don't know in which cases the error occurs, and I wasn't able to reproduce it.

**Possible Solution**
I think we should handle non-array return types of the `unserialize` method.

**Additional context**

Method : https://github.com/symfony/symfony/blob/5.4/src/Symfony/Component/HttpKernel/HttpCache/Store.php#L295

```
{
    "class": "Symfony\\Component\\Debug\\Exception\\FatalThrowableError",
    "message": "Return value of Symfony\\Component\\HttpKernel\\HttpCache\\Store::getMetadata() must be of the type array, bool returned",
    "code": 0,
    "file": "vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpCache/Store.php:301",
    "trace": [
        "vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpCache/Store.php:134",
        "vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpCache/HttpCache.php:330",
        "vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpCache/HttpCache.php:222",
        "vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpCache/AbstractSurrogate.php:96",
        "vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpCache/HttpCache.php(637) : eval()'d code:360",
        "vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpCache/HttpCache.php:637",
        "vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpCache/HttpCache.php:225",
        "public/index.php:31"
    ]
}
```

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #42031
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Component/HttpKernel/HttpCache/Store.php [hunk 1]
```diff
@@ -298,7 +298,7 @@ private function getMetadata(string $key): array
             return [];
         }
 
-        return unserialize($entries);
+        return unserialize($entries) ?: [];
```
Comment (EmmanuelVella) [inline src/Symfony/Component/HttpKernel/HttpCache/Store.php:301]: `unserialize` can return various types (see https://www.php.net/manual/en/function.unserialize.php#refsect1-function.unserialize-returnvalues).

Maybe we should test the return value with `is_array` ?

Comment (stof) [reply src/Symfony/Component/HttpKernel/HttpCache/Store.php:301]: Well, the Store controls the serialization too. So if unserializing does not fail, we know that we have an array (unless you messed up with the files directly to change what is stored, but that's not supported by our implementation)

**Comment by nicolas-grekas** [comment]
Thank you @xabbuh.

**Comment by EmmanuelVella** [comment]
Thank you @xabbuh !

**Pull request merged**